### ISSUE TITLE
How can I from A UILabel to B UILabel with different colors?

### ISSUE THREAD

**Issue by pxwinter** [HeroTransitions/Hero#414]
Hi 
How can I from A Label to B Label with different colors?

For example
A Label with red color in VC1, and B Label with blue color in VC2, how can I translate them by Hero.

Thanks and best regards

**Comment by lkzhao** [comment]
Not sure if I understand your question clearly. But setting the heroId to be the same for both label should do it right?

**Comment by pxwinter** [comment]
@lkzhao Thanks for your feedback, could you please look at this attach file, I record a  demo video.
In the progress the animation, the red label didn't fade away. How can I add a fade animation on red label (opacity from 1 to 0)
![demo1](https://user-images.githubusercontent.com/874482/36883563-2f0a079c-1e16-11e8-93a8-8ca3af5f180c.gif)

Thanks and best regards

**Comment by brokenzz** [comment]
@pxwinter I came across a similar issue whereby:

- if my destination ViewController is instantiated via Storyboard, this scenario of old text artifact being visible throughout duration is not present
- however, if I recreate the same ViewController manually, and set it up in code the same way it is setup in storyboard, then I would get the same result you posted.

After some debugging, the solution I found is simple. In Storyboard when you add a UILabel, Xcode by default set's it's `isOpaque` property to `false`.

However, if you create a UILabel in code, it's `isOpaque` property for some reason is set to `true`.

In Hero's `MatchPreprocessor` is appears to behave differently depending on whether the view is opaque or not, on line `44`:
```swift
let isNonOpaque = !fv.isOpaque || fv.alpha < 1 || !tv.isOpaque || tv.alpha < 1
```
So to solve my problem it was enough to simply set `label.isOpaque = false` on the **destination** label.
